plutus-core/plutus-ir/test/transform/inline/letFunInFunMultiFullyApplied.golden

```diff
@@ -0,0 +1,9 @@
+(let
+  (nonrec)
+  (termbind
+    (strict)
+    (vardecl idFun (fun (con integer) (con integer)))
+    (lam x (con integer) x)
+  )
+  [ [ (lam y (fun (con integer) (con integer)) idFun) idFun ] (con integer 1) ]
```

Comment:
Neither of the two occurrences of `idFun` is fully applied, so I don't think it will be inlined.

Reply:
Sorry I don't see why they are not fully applied. First application is
```[ (lam y (fun (con integer) (con integer)) idFun) idFun ]``` in Haskell ```\y.idFun idFun``` so it simplifies to `idFun`.

Second one is what the first simplifies to (i.e. `idFun`) applies to `(con integer 1)`.

Reply:
> In Haskell: \y.idFun idFun so it simplifies to idFun.

Like @michaelpj said, in order to simplify `(\y -> idFun) idFun` to `idFun`, you need to run beta reduction. And if you do run beta reduction to simplify `(\y -> idFun) idFun` to `idFun`, then `idFun` will now have a single occurrences, and will be unconditionally inlined. So this cannot test inlining fully saturated functions.

Reply:
hm...I'll think about whether I want to delete this or somehow make it not single occurrence then. Any preference? Maybe I can just delete this?

Reply:
You can just change `(con integer 1)` to `[ idFun (con integer 1) ]`. Then this `idFun` occurrence will be inlined, the other two won't.

Reply:
I think it helps to focus on the fact that these are tests _just_ for the inliner. That's the only thing that we run! So in particular:
- You can't assume other simplification will happen, that's not what we're doing
- Dead code won't run

The second bit means that you can easily make things multiple-occurrence by just adding references to them in dead code, e.g.
```
let x = ..
      y = x
in ... x ...
```

Reply:
Yes, I know that's what you mean about dead code, but remember? The unconditional inliner has deadcode removal in it! See [nameUsedAtMostOnce](https://github.com/input-output-hk/plutus/blob/48c326a1c155be44b67a366241eb29c80e3beed2/plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Utils.hs#L161).